Specs/Scene/ImageryLayerCollectionSpec.js

```diff
@@ -258,6 +258,260 @@ describe(
+        var provider = {
+          ready: true,
+          rectangle: Rectangle.MAX_VALUE,
+          tileWidth: 256,
+          tileHeight: 256,
+          maximumLevel: 0,
+          minimumLevel: 0,
+          tilingScheme: new GeographicTilingScheme(),
+          errorEvent: new Event(),
+          hasAlphaChannel: true,
+
+          requestImage: function (x, y, level) {
+            return ImageryProvider.loadImage(this, "Data/Images/Blue.png");
+          },
+        };
+
+        globe.imageryLayers.addImageryProvider(provider);
+
+        return updateUntilDone(globe).then(function () {
+          var ellipsoid = Ellipsoid.WGS84;
+          camera.lookAt(
+            new Cartesian3(ellipsoid.maximumRadius, 0.0, 0.0),
+            new Cartesian3(0.0, 0.0, 100.0)
+          );
+
+          var ray = new Ray(camera.position, camera.direction);
+          var imagery = scene.imageryLayers.pickImageryLayers(ray, scene);
+          expect(imagery).toBeUndefined();
+        });
+      });
+
+      it("returns undefined when ImageryProvider.pickFeatures returns undefined", function () {
+        var provider = {
+          ready: true,
+          rectangle: Rectangle.MAX_VALUE,
+          tileWidth: 256,
+          tileHeight: 256,
+          maximumLevel: 0,
+          minimumLevel: 0,
+          tilingScheme: new GeographicTilingScheme(),
+          errorEvent: new Event(),
+          hasAlphaChannel: true,
+
+          pickFeatures: function (x, y, level, longitude, latitude) {
+            return undefined;
+          },
+
+          requestImage: function (x, y, level) {
+            return ImageryProvider.loadImage(this, "Data/Images/Blue.png");
+          },
+        };
+
+        globe.imageryLayers.addImageryProvider(provider);
+
+        return updateUntilDone(globe).then(function () {
+          var ellipsoid = Ellipsoid.WGS84;
+          camera.lookAt(
+            new Cartesian3(ellipsoid.maximumRadius, 0.0, 0.0),
+            new Cartesian3(0.0, 0.0, 100.0)
+          );
+
+          var ray = new Ray(camera.position, camera.direction);
+          var imagery = scene.imageryLayers.pickImageryLayers(ray, scene);
+          expect(imagery).toBeUndefined();
+        });
+      });
+
+      it("returns imagery from one layer", function () {
+        var provider = {
+          ready: true,
+          rectangle: Rectangle.MAX_VALUE,
+          tileWidth: 256,
+          tileHeight: 256,
+          maximumLevel: 0,
+          minimumLevel: 0,
+          tilingScheme: new GeographicTilingScheme(),
+          errorEvent: new Event(),
+          hasAlphaChannel: true,
+
+          pickFeatures: function (x, y, level, longitude, latitude) {
+            var deferred = when.defer();
+            setTimeout(function () {
+              var featureInfo = new ImageryLayerFeatureInfo();
+              deferred.resolve([featureInfo]);
+            }, 1);
+            return deferred.promise;
+          },
```

Comment:
Same comment: `pickFeatures` is not required for `pickImageryLayers` to work. It can be removed from these new tests.